### PR TITLE
Instructions for using rescript-mode with eglot in Doom Emacs

### PR DESCRIPTION
**Pull request by glennsl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,39 @@ shouldn't be too hard to rewrite this to not use `use-package`.
           "--stdio"))))
 ```
 
+### Doom Emacs
+
+#### [eglot](https://github.com/joaotavora/eglot)
+
+In `init`el`, ensure you have the `+eglot` option for `lsp`:
```
Comment (jjlee) [inline README.md]: init.el?

File: README.md [hunk 2]
```diff
@@ -95,6 +95,39 @@ shouldn't be too hard to rewrite this to not use `use-package`.
...
+(package! rescript-mode)
+```
+
+Then in `config.el` add:
+
+```elisp
+(after! eglot
+  (add-to-list 'eglot-server-programs
+               '(rescript-mode . ("node"
+                                  "/path/to/rescript-vscode/server/out/server.jss"
```
Comment (jjlee) [inline README.md]: server.js?

File: README.md [hunk 3]
```diff
@@ -95,6 +95,39 @@ shouldn't be too hard to rewrite this to not use `use-package`.
...
+                                  "--stdio")))
+  )
+
+(add-hook 'rescript-mode-hook (lambda ()
+                                (eglot-ensure)
+                                (electric-indent-local-mode -1)
```
Comment (jjlee) [inline README.md]: Maybe something to explain why disabling `elecric-indent-mode`?

Comment (glennsl) [reply README.md]: Ah, sorry, it's not needed. I just disable it because it's annoying. Will remove it.

**Pull request merged**